File: HashTable/217. Contains Duplicate.py
class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        #just checking the length is faster but more memory
        return len(set(nums))!=len(nums)
        #hashsets don't use much memory but it's not fast
        hashset = set()
        for x in nums:
            if x in hashset:
                return True
            else:
                hashset.add(x)
        return False
        

File: HashTable/217. Contains Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        //find duplicates using hashset
        Set<Integer> hashset = new HashSet<>();
        for (int key : nums) {
            if (hashset.contains(key)) {
                return true;
            }
            hashset.add(key);
        }
        return false;
        
    }
}